client/db/src/lib.rs

```diff
@@ -1399,6 +1498,49 @@ impl<Block: BlockT> Backend<Block> {
 
 		Ok(())
 	}
+
+	fn prune_blocks(
+		&self,
+		transaction: &mut Transaction<DbHash>,
+		finalized: NumberFor<Block>,
+	) -> ClientResult<()> {
+		if let KeepBlocks::Some(keep_blocks) = self.keep_blocks {
+			// Always keep the last finalized block
+			let keep = std::cmp::max(keep_blocks, 1);
+			if finalized >= keep.into() {
+				let number = finalized.saturating_sub(keep.into());
+				match read_db(&*self.storage.db, columns::KEY_LOOKUP, columns::BODY, BlockId::<Block>::number(number))? {
+					Some(body) => {
```

Comment:
Is there a use case where we want to remove the transactions but not the body reference? (from description not really)

Reply:
Don't think so